jenkins/shared-library/vars/codeCoverage.groovy

```diff
@@ -43,6 +43,14 @@ void verifyCodeCoverageResult(String tool, Integer minimumCodeCoverage) {
 		'jacoco': { Integer target ->
 			logger.logInfo("Minimum code coverage set in pom is ${readJacocoCoverageLimit()}")
 			runScript('mvn jacoco:check@jacoco-check')
+		},
+		'golang': { Integer target ->
+			logCodeCoverageMinimum(target)
+			runScript('go tool cover -func coverage.out')
+			String coverage = runScript("go tool cover -func coverage.out | grep total | awk '{print substr(\$3, 1, length(\$3)-3)}'", true)
```

Comment:
are you running coverage twice ?

Reply:
This analyzes the coverage profiles generated by the test run to create coverage percentages.

Reply:
i mean you are running script `go tool cover -func coverage.out` two times, is that intentional ?

Reply:
yea, didnt want write code to parse the output 😅 

Will update 🙏🏾